src/test/java/org/example/Operations/CollectionOperationsTest.java

```diff
@@ -1,250 +1,483 @@
+        assertEquals("Creating collection failed", 0,
+                getIrodsResponseStatusCode(res.getBody()));
+
+        // Stat the collection to show that it exists.
+        res = rods.collections().stat(token, collectionPath, Optional.empty());
+        assertEquals("Stat request failed", 200, res.getHttpStatusCode());
+        assertEquals("Stat failed", 0,
+                getIrodsResponseStatusCode(res.getBody()));
+
+        // Rename the collection.
+        String newCollectionPath = collectionPath + ".renamed";
+        res = rods.collections().rename(token, collectionPath, newCollectionPath);
+        assertEquals("Renaming collections request failed", 200, res.getHttpStatusCode());
+        assertEquals("Renaming collections failed", 0,
+                getIrodsResponseStatusCode(res.getBody()));
+
+        // Stat the original collection to show that it does not exist.
+        res = rods.collections().stat(token, collectionPath, Optional.empty());
+        assertEquals("Stat request failed", 200, res.getHttpStatusCode());
+        assertEquals("Stat did not fail as expected", -170000,
+                getIrodsResponseStatusCode(res.getBody()));
+        // -170000: NOT_A_COLLECTION
+
+        // Stat the new collection to show that it does exist.
+        res = rods.collections().stat(token, newCollectionPath, Optional.empty());
+        assertEquals("Stat request failed", 200, res.getHttpStatusCode());
+        assertEquals("Stat failed", 0,
+                getIrodsResponseStatusCode(res.getBody()));
+
+        // Give another user (alice) permission to read the object.
+        res = rods.collections().set_permission(token, newCollectionPath, "alice", Permission.READ,
+                OptionalInt.empty());
+        assertEquals("Setting permission request failed", 200, res.getHttpStatusCode());
+        assertEquals("Setting permission failed", 0,
+                getIrodsResponseStatusCode(res.getBody()));
+
+        // Show that the alice user now has read permission on the collection.
+        res = rods.collections().stat(token, newCollectionPath, Optional.empty());
+        assertEquals("Stat request failed", 200, res.getHttpStatusCode());
+        assertEquals("Stat failed", 0,
+                getIrodsResponseStatusCode(res.getBody()));
+
+        JsonNode rootNode;
+        try {
+            rootNode = mapper.readTree(res.getBody());
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+        JsonNode permissionsNode = rootNode.path("permissions");
+        boolean permissionExists = false;
+        for (JsonNode permission : permissionsNode) {
+            if (permission.path("name").asText().equals("alice") &&
+                    permission.path("perm").asText().equals("read_object") &&
+                    permission.path("type").asText().equals("rodsuser") &&
+                    permission.path("zone").asText().equals("tempZone")) {
```

Comment:
To avoid null pointer exceptions, swap the operands. For example:
```java
"name".equals(permission.path("name").asText()
```

Can `.path("<string>")` return a null reference?
Can `.asText()` return a null reference?

Reply:
I swapped the operands. 

`.path("<string>")` returns a "missing node" and `.asText()` returns an empty string.

https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/2.8.5/com/fasterxml/jackson/databind/JsonNode.html

Reply:
Yep